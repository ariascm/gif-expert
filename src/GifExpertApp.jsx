import { useState } from "react"
import { AddCategory, GifGrid } from "./components"

export const GifExpertApp = () => {

    const [categories, setCategories] = useState(['Batman'])

    const onAddCategory = (newCategory) => {
        if (categories.includes(newCategory)) return;

        // NO usar PUSH, ya que no modificamos la constante, creamos un nuevo objeto
        setCategories([newCategory, ...categories])
    }

    return (
        <>
            <h1>GifExpertApp</h1>

            <AddCategory
                onNewCategory={(value) => onAddCategory(value)}
            />

            {
                categories.map(category => (
                    <GifGrid
                        key={category}
                        category={category}
                    />
                ))
            }

        </>
    )
}
