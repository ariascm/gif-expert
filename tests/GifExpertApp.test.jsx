import { fireEvent, render, screen } from "@testing-library/react"
import { GifExpertApp } from "../src/GifExpertApp"

describe('Pruebas en <GifExpertApp/>', () => {


    test('Debe hacer match con el snapshot', () => {

        const { container } = render(<GifExpertApp />)
        expect(container).toMatchSnapshot()

        // screen.debug()
    })

    test('Submitear dos veces la misma categoria y que se muestre una sola vez', () => {

        const nuevoValor = "Marvel"

        render(<GifExpertApp />)
        const input = screen.getByRole('textbox')

        fireEvent.input(input, { target: { value: nuevoValor } })

        expect(input.value).toBe(nuevoValor)

        const form = screen.getByRole('form')
        fireEvent.submit(form)

        expect(screen.getByText(nuevoValor)).toBeTruthy()

        fireEvent.input(input, { target: { value: nuevoValor } })
        fireEvent.submit(form)
        // screen.debug()
        expect(screen.getAllByText(nuevoValor).length).toBe(1)
    })

})