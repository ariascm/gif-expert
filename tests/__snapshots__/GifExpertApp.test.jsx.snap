// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifExpertApp/> Debe hacer match con el snapshot 1`] = `
<div>
  <h1>
    GifExpertApp
  </h1>
  <form
    aria-label="formLabel"
  >
    <input
      placeholder="Buscar gifs"
      type="text"
      value=""
    />
  </form>
  <h3>
    Batman
  </h3>
  <h4>
    cargando...
  </h4>
  <div
    class="card-grid"
  />
</div>
`;
